hack/run-locally.sh

```diff
@@ -94,24 +121,184 @@ function wait_for_cluster() {
+                    *)
+                        echo "Unknown network plugin ${OPTARG}" >&2
+                        exit 1
+                    ;;
+                esac
+                ;;
+            \?)
+                echo "Invalid option: -${OPTARG}" >&2
+                exit 1
+                ;;
+            :)
+                echo "Option -${OPTARG} requires an argument." >&2
+                exit 1
+                ;;
+            *)
+                print_usage
+                exit 2
+                ;;
+        esac
+    done
+    ;;
+
+    *)
+        print_usage
+        exit 2
+        ;;
+esac
+
 if [[ -z "${CLUSTER_DIR:-}" ]]; then
-    echo "error: CLUSTER_DIR must be set"
+    echo "error: CLUSTER_DIR must be set or '-c <cluster-dir>' must be given"
     echo "For more info, see INSTALLER-HACKING.md"
     exit 1
 fi
+mkdir -p "${CLUSTER_DIR}" >& /dev/null
 
-if [[ ! -e "${CLUSTER_DIR}/manifests" && ! -e "${CLUSTER_DIR}/auth" ]]; then
-    echo "error: cannot find installer state; please run"
-    echo "  openshift-install --dir=${CLUSTER_DIR} create manifests"
-    echo "For more info, see INSTALLER-HACKING.md"
+if [[ -z "$(which oc 2> /dev/null || exit 0)" ]]; then
+    echo "could not find 'oc' in PATH" >&2
     exit 1
 fi
 
-if [[ -e "${CLUSTER_DIR}/manifests/cvo-overrides.yaml" ]]; then
-    override_install_manifests
+if [[ "${MODE}" == "new" ]]; then
+    # Don't clobber an existing cluster data directory
+    if [[ -n "$(ls -A ${CLUSTER_DIR} | grep -v install-config.yaml | grep -v .openshift_install.log)" ]]; then
+        echo "cluster-dir not empty; will not overwrite existing cluster" >&2
+        exit 1
+    fi
+
+    # Find openshift-install if not explicitly given
+    if [[ -z "${INSTALLER_PATH}" ]]; then
+        INSTALLER_PATH="$(which openshift-installer 2> /dev/null || exit 0)"
+        if [[ -z "${INSTALLER_PATH}" ]]; then
+            echo "could not find openshift-installer in PATH to build from-scratch cluster" >&2
+            exit 1
+        fi
+    fi
+
+    if [[ -z "${INSTALL_CONFIG}" ]]; then
+        # Create a new install-config.yaml if none was given and none exists in CLUSTER_DIR
+        if [[ ! -f "${CLUSTER_DIR}/install-config.yaml" ]]; then
+            echo "Creating install config..."
+            "${INSTALLER_PATH}" --dir="${CLUSTER_DIR}" create install-config
+        fi
+    else
+        cp "${INSTALL_CONFIG}" "${CLUSTER_DIR}/install-config.yaml"
+    fi
+
+    # Update the requested network plugin
+    sed -i -e "s/networkType: [a-zA-Z]*$/networkType: $NETWORK_PLUGIN/" "${CLUSTER_DIR}/install-config.yaml"
+
+    echo "Creating install manifests..."
+    "${INSTALLER_PATH}" --dir="${CLUSTER_DIR}" create manifests
+fi
+
+if [[ "${MODE}" != "attach" ]]; then
+    if [[ ! -e "${CLUSTER_DIR}/manifests" && ! -e "${CLUSTER_DIR}/auth" ]]; then
+        echo "error: cannot find installer state; please run"
+        echo "  openshift-install --dir=${CLUSTER_DIR} create manifests"
```

Comment:
`create cluster` would be better, right?

Reply:
@squeed actually no, because we create-cluster later in the 'new' case below, and at this point all we really need are the manifests.